modules/ROOT/pages/policy-openid-connect.adoc

```diff
@@ -8,98 +8,182 @@ endif::[]
 
-Token validation response example from the authorization server:
+== Configuring the Expose Headers Option
+
+If you configured the `Expose Headers` option in the policy, the application (for example, out-of-the-box proxies) uses the HTTP requester to redirect the properties to the backend as headers. 
+
+For each field, the original request to the protected resource is enriched with the HTTP headers using the format: X-AGW- + <key>=<value>. For the example response from the previous section, the following headers will be added:
 
 [source,text,linenums]
-----
-{
-    "uid":"john.doe",
-    "mail":"john.doe@example.com",
-    "scope":["uid","mail","cn","givenName"],
-    "grant_type":"password",
-    "cn":"John Doe",
-    "realm":"/",
-    "token_type":"Bearer",
-    "expires_in":3600,
-    "givenName":"John",
-    "access_token":"fa017a0e-1bd5-214c-b19d-03efe9f9847e"
-}
-----
+---
+X-AGW-uid=john.doe
+X-AGW-mail=john.doe@example.com
+X-AGW-token_type=Bearer
+X-AGW-expires_in=3600
+---
+
+For further processing within the flow, the original unparsed response from the validation endpoint is made available. The information is stored in an authentication object. 
+
+For example, suppose a `"mail`" field is returned to the policy, you can access the value of the field by using the following DataWeave 2.0 expression:
+
+`#[authentication.properties.userProperties.mail]`
+
+You can obtain the client ID of the OAuth2 token by using the following DataWeave 2.0 expression:
+
+`#[authentication.principal]`
+
+== Communicating With the Token Validation Endpoint Using a Proxy
+
+You can enable the OpenID Connect OAuth 2.0 Token Enforcement policy to use the gateway proxy settings by specify the following property when starting the Mule runtime engine: 
+
+`anypoint.platform.external_authentication_provider_enable_proxy_settings=<true|false(default)>`
+
+When you enable this property, the policy uses the Mule proxy settings, if you have specified the following parameters:
+
+* `anypoint.platform.proxy_host=localhost`
+* `anypoint.platform.proxy_port=8080`
+
+== Configuring Policy Parameters
+
+When you apply the policy to your API from the UI, a list of parameters are displayed based on whether your environment includes Mule or xref:service-mesh::index.adoc[Non-Mule application managed by Anypoint Service Mesh].
+
+=== Configuring the Parameters for Mule Applications
+
+For Mule applications, the following parameters are displayed based on whether your environment includes Mule applications:
+
+[%header%autowidth.spread,cols="30,40,30"]
+|===
+| Element | Description | Example
+| Scopes | A space-separated list of supported scopes. | READ, WRITE, READ and WRITE 
+| Scope Validation Criteria | Determines if the token must contain all defined scopes or just one.
+Values: `Contains all scopes` and `Contains any scope` | 
+In this example, the value `READ,WRITE` was selected for the scope element.
```

Comment:
Which example?

Reply:
Is READ,WRITE consistent with the values listed in Line 125?

Reply:
past tense

Reply:
Done! Thanks!